File: static/include/examples-generated/data-region.snippet.data-region.ts
import { createViamClient } from "@viamrobotics/sdk";

// Configuration constants – replace with your actual values
let API_KEY = "";  // API key, find or create in your organization settings
let API_KEY_ID = "";  // API key ID, find or create in your organization settings
let ORG_ID = "";  // Organization ID, find or create in your organization settings

async function main(): Promise<void> {
    // Create Viam client
    const client = await createViamClient({
        credentials: {
            type: "api-key",
            authEntity: API_KEY_ID,
            payload: API_KEY,
        },
    });

    // Check organization region
    const org = await client.appClient.getOrganization(ORG_ID);
    console.log(`Current region: ${org.defaultRegion}`);

    // Update organization region
    try {
        const updatedOrg = await client.appClient.updateOrganization(ORG_ID, "us-central");
        console.log(`Organization region updated to: ${updatedOrg?.defaultRegion}`);
    } catch (e) {
        console.log(`Error updating organization region: ${e}`);
    }
}

// Run the script
main().catch((error) => {
    console.error("Script failed:", error);
    process.exit(1);
});

File: static/include/examples-generated/data-region.snippet.data-region.py
import asyncio
import os

from viam.rpc.dial import DialOptions, Credentials
from viam.app.viam_client import ViamClient

# Configuration constants – replace with your actual values
API_KEY = ""  # API key, find or create in your organization settings
API_KEY_ID = ""  # API key ID, find or create in your organization settings
ORG_ID = ""  # Organization ID, find or create in your organization settings


async def connect() -> ViamClient:
    """Establish a connection to the Viam client using API credentials."""
    dial_options = DialOptions(
        credentials=Credentials(
            type="api-key",
            payload=API_KEY,
        ),
        auth_entity=API_KEY_ID
    )
    return await ViamClient.create_from_dial_options(dial_options)


async def main() -> int:
    viam_client = await connect()

    # Check organization region
    org = await viam_client.app_client.get_organization(org_id=ORG_ID)
    print(f"Current region: {org.default_region}")

    # Update organization region
    try:
        updated_org = await viam_client.app_client.update_organization(
            org_id=ORG_ID,
            region="us-central"  # or "us-central"
        )
        print(f"Organization region updated to: {updated_org.region}")
    except Exception as e:
        print(f"Error updating organization region: {e}")

    viam_client.close()
    return 0

if __name__ == "__main__":
    asyncio.run(main())
